File: examples_ios/MyApp/Sources/SurveyLayouts/DefaultLayout/Items/DefaultLayoutItemUtils.swift
import Foundation
import ConfirmitMobileSDK

enum DefaultLayoutItemUtils {
    static func getSingleAnswerIndexList(section: Int, question: SingleQuestion, excludeCode: String) -> [IndexPath] {
        var startIndex = 0
        if !question.title.isEmpty { startIndex += 1 }
        if !question.text.isEmpty { startIndex += 1 }
        if !question.instruction.isEmpty { startIndex += 1 }
        if question.errors.count > 0 { startIndex += 1 }
        
        var result = [IndexPath]()
        for answer in question.answers {
            if !answer.isHeader && answer.code != excludeCode {
                result.append(IndexPath(row: startIndex, section: section))
            }
            
            if answer.isHeader {
                if !answer.text.isEmpty {
                    startIndex += 1
                }
            } else {
                startIndex += 1
            }
            
            for _ in answer.answers {
                if answer.code != excludeCode {
                    result.append(IndexPath(row: startIndex, section: section))
                }
                
                startIndex += 1
            }
        }
        
        return result
    }
    
    static func getValidationIndex(section: Int, question: DefaultQuestion) -> IndexPath {
        var startIndex = 0
        if !question.title.isEmpty { startIndex += 1 }
        if !question.text.isEmpty { startIndex += 1 }
        if !question.instruction.isEmpty { startIndex += 1 }
        return IndexPath(row: startIndex, section: section)
    }
}

File: examples_android/app/src/main/java/com/example/myapp/layouts/defaultlayout/items/DefaultLayoutItemUtils.kt
package com.example.myapp.layouts.defaultlayout.items

import com.confirmit.mobilesdk.ui.questions.DefaultQuestion
import com.confirmit.mobilesdk.ui.questions.SingleQuestion

class IndexPath(var row: Int, var section: Int)

object DefaultLayoutItemUtils {
    fun getSingleAnswerIndexList(section: Int, question: SingleQuestion, excludeCode: String): List<IndexPath> {
        var startIndex = 0
        if (!question.title.isEmpty()) startIndex += 1
        if (!question.text.isEmpty()) startIndex += 1
        if (!question.instruction.isEmpty()) startIndex += 1
        if (question.errors.isNotEmpty()) startIndex += 1

        val result = mutableListOf<IndexPath>()
        question.answers.forEach { answer ->
            if (!answer.isHeader && answer.code != excludeCode) {
                result.add(IndexPath(startIndex, section))
            }

            if (answer.isHeader) {
                if (!answer.text.isEmpty()) {
                    startIndex += 1
                }
            } else {
                startIndex += 1
            }

            for (nestedAnswer in answer.answers) {
                if (answer.code != excludeCode) {
                    result.add(IndexPath(startIndex, section))
                }

                startIndex += 1
            }
        }

        return result
    }

    fun getValidationIndex(section: Int, question: DefaultQuestion): IndexPath {
        var startIndex = 0
        if (!question.title.isEmpty()) startIndex += 1
        if (!question.text.isEmpty()) startIndex += 1
        if (!question.instruction.isEmpty()) startIndex += 1
        return IndexPath(startIndex, section)
    }
}